contrib/inventory/ec2.py

```diff
@@ -672,7 +672,9 @@ def get_elasticache_clusters_by_region(self, region):
                 error = self.get_auth_error_message()
             if not e.reason == "Forbidden":
                 error = "Looks like AWS ElastiCache is down:\n%s" % e.message
-            self.fail_with_error(error, 'getting ElastiCache clusters')
+                self.fail_with_error(error, 'getting ElastiCache clusters')
+            else:
+                return
```

Comment:
I consider that a violation of least surprise.  if you have your ec2_inventory set to get elasticache or rds instances and it returns none due to error when in fact there are instances, that is surprising.